### PR TITLE
[css-display-3] Tests for display: contents.

### PR DESCRIPTION
**Pull request by emilio**
This imports an splitted version of mozilla's [display: contents acid test](http://searchfox.org/mozilla-central/source/layout/reftests/css-display/display-contents-acid.html), and adds tests for dynamic changes to display: contents elements in a variety of situations both from and to `display: none` and `display: inline`.

I'm aware that there's a bunch of duplication, though adding a bunch of iframes for this seemed overkill. Let me know if you think it's fine, or otherwise what's the best way to move this forward. I really like more minimal tests, but these tests test a variety of hard-to-find-otherwise situations.

Could you take a look @rune-opera? Thanks!

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/csswg-test/1155)

### CODE REVIEW

File: css-display-3/display-contents-dynamic-flex-001-none.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<!doctype html>
```
Comment (rune-opera) [inline css-display-3/display-contents-dynamic-flex-001-none.html]: Use upper-case for DOCTYPE

File: css-display-3/display-contents-dynamic-flex-001-inline.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<!doctype html>
```
Comment (rune-opera) [inline css-display-3/display-contents-dynamic-flex-001-inline.html]: Use upper-case for DOCTYPE

File: css-display-3/display-contents-dynamic-flex-001-inline.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  display: inline;
+}
+.contents {
+  display: contents;
+}
+</style>
+<div class="flex">
+0
+  <div class="contents">
+    <div>1</div>
```
Comment (rune-opera) [inline css-display-3/display-contents-dynamic-flex-001-inline.html]: You should be able to see if the test passed or not by having a clear pass condition in the test case without having to look at the reference. That would make the intention of the test clearer too.

File: css-display-3/display-contents-text-only-001.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Display: Only text on a display: contents subtree</title>
+<link rel="author" title="Emilio Cobos Álvarez" href="mailto:ecobos@igalia.com">
+<link rel="help" href="https://drafts.csswg.org/css-display-3/#valdef-display-contents">
+<link rel="match" href="display-contents-text-only-001-ref.html">
+<style>
+  html, body { display: contents; }
+</style>
+PASS
```
Comment (rune-opera) [inline css-display-3/display-contents-text-only-001.html]: Are you really allowed to have display:contents on html (or other root elements)? We should have a test for that.

**Comment by rune-opera** [comment]
In general, I think we should have clear pass conditions in the test-cases.

You've credited Mozilla authors in some of the tests. Did you add credits to all tests from Mozilla?

File: css-display-3/display-contents-text-only-001.html (hunk 4 above)
Comment (emilio) [inline css-display-3/display-contents-text-only-001.html]: Hmm, good catch. I've allowed it in my impl, but after double-checking Gecko computes the document element's `display` to `block`. This is not spec'd, but feels sane.

There are tests that specify this, though (which is why I added this test case): https://github.com/w3c/web-platform-tests/blob/master/html/semantics/grouping-content/the-li-element/grouping-li-reftest-no-list-owner.html

Should I:

 * Open a spec issue asking for clarification, or
 * Change my implementation to match Gecko's and add a test for it, or
 * File a Gecko bug, keep this test, and write an additional test for the computed display of the document element?

I'm fine with any of them.

File: css-display-3/display-contents-dynamic-flex-001-none.html (hunk 1 above)
Comment (emilio) [inline css-display-3/display-contents-dynamic-flex-001-none.html]: Good catch.

File: css-display-3/display-contents-dynamic-flex-001-inline.html (hunk 3 above)
Comment (emilio) [inline css-display-3/display-contents-dynamic-flex-001-inline.html]: I will try on the tests created by me.

File: css-display-3/display-contents-dynamic-flex-001-inline.html (hunk 2 above)
Comment (emilio) [inline css-display-3/display-contents-dynamic-flex-001-inline.html]: Good catch, again. Thanks :)

**Comment by emilio** [comment]
I kept the fixup commits intentionally so it's easier to review, I'll squash when appropriate :)

**Comment by emilio** [comment]
I added tests for the "blockification" of the document element, and also for floats and out-of-flows that get coerced to blocks in other contexts, but should not do so for `display: contents`.

**Comment by emilio** [comment]
@rune-opera ping?

**Comment by rune-opera** [comment]
There are still test cases without clear pass conditions.

**Comment by emilio** [comment]
So as I said I did so on the tests created by me.

I'll modify also Mozilla's tests, I guess, though the complexity of those tests is enough that rewriting those so passing them is clear is non-trivial.

**Comment by rune-opera** [comment]
This is the first time I review csswg-tests. Perhaps someone with more experience in doing so should do this. I don't think the fact that they are existing external tests should relax the requirements for them. If they are too complex, they should possibly be rewritten to be simpler, even if that is more work.

**Comment by emilio** [comment]
@gsnedders opinions?

Context: In this PR I imported several tests from Gecko. I agree with Rune that it'd be desirable to get tests that test exactly the same thing as those, but with clear passing conditions. That's a fair amount of work, though.

I'm fine with whatever decision ends up being taken, though I personally don't think rewriting those tests is worth the effort.

**Comment by emilio** [comment]
If we go for a rewrite, I guess I'll remove those tests from this PR, so the rest of them can be merged, and then submit them separately.

**Comment by gsnedders** [comment]
I haven't read through all of @rune-opera's comments (soz!), but if this is about having pass conditions: we don't require this for anything using `testharness.js` (as the harness passes back PASS/FAIL and typically displays it), reftests (because ultimately they have a programmatically tested pass condition, though we do prefer it), but do for visual and manual tests.

**Comment by emilio** [comment]
I guess this is good to land then then if @rune-opera doesn't have any objection?

**Comment by rune-opera** [comment]
*[css-display-3/display-contents-text-only-001.html, line 10 at r2](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KXqX0UyF_2vou0l3RcP-r2-10:-KYwW-Puwbt0jk4KPCwT:b25jtlf) ([raw file](https://github.com/w3c/csswg-test/blob/3be5c4bd72336a0ab37547d0293d2463ddeb1060/css-display-3/display-contents-text-only-001.html#L10)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary><blockquote>

Hmm, good catch. I've allowed it in my impl, but after double-checking Gecko computes the document element's `display` to `block`. This is not spec'd, but feels sane.

There are tests that specify this, though (which is why I added this test case): https://github.com/w3c/web-platform-tests/blob/master/html/semantics/grouping-content/the-li-element/grouping-li-reftest-no-list-owner.html

Should I:

 * Open a spec issue asking for clarification, or
 * Change my implementation to match Gecko's and add a test for it, or
 * File a Gecko bug, keep this test, and write an additional test for the computed display of the document element?

I'm fine with any of them.

</blockquote></details>

Yes, we need a spec issue at least. Since Blink won't ship yet, and this is a corner case, I don't think it's important what we do in the implementation as long as we change to what the spec issue resolves to before shipping.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by emilio** [comment]
Review status: 0 of 39 files reviewed at latest revision, 1 unresolved discussion.

---

*[css-display-3/display-contents-text-only-001.html, line 10 at r2](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KXqX0UyF_2vou0l3RcP-r2-10:-KYwftcYy3mZizS4xxRN:b-qrocmm) ([raw file](https://github.com/w3c/csswg-test/blob/3be5c4bd72336a0ab37547d0293d2463ddeb1060/css-display-3/display-contents-text-only-001.html#L10)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

Yes, we need a spec issue at least. Since Blink won't ship yet, and this is a corner case, I don't think it's important what we do in the implementation as long as we change to what the spec issue resolves to before shipping.
</blockquote></details>

Filled https://github.com/w3c/csswg-drafts/issues/805

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by rune-opera** [comment]
*[css-display-3/display-contents-text-only-001.html, line 10 at r2](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KXqX0UyF_2vou0l3RcP-r2-10:-KYwkIOV-8j6OFfnDNx5:bj61xgt) ([raw file](https://github.com/w3c/csswg-test/blob/3be5c4bd72336a0ab37547d0293d2463ddeb1060/css-display-3/display-contents-text-only-001.html#L10)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary><blockquote>

Filled https://github.com/w3c/csswg-drafts/issues/805
</blockquote></details>

I resolved it as blockification of contents on root element is specified in css-display-3.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by rune-opera** [comment]
Reviewed 1 of 33 files at r1, 2 of 6 files at r3, 1 of 1 files at r4.
Review status: 4 of 39 files reviewed at latest revision, 6 unresolved discussions.

---

*[css-display-3/display-contents-computed-style.html, line 72 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwkfbGJvD0lzkhHBnq:-KYwkfbGJvD0lzkhHBnr:b-vth342) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-computed-style.html#L72)):*
> ```HTML
>     test(function(){
>         assert_equals(getComputedStyle(document.documentElement).display, "block");
>     }, "Document element can't have display: contents");
> ```

Add a spec link to https://drafts.csswg.org/css-display-3/#transformations and -> "display:contents is blockified for root elements"

---

*[css-display-3/display-contents-dynamic-flex-001-inline.html, line 22 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwpXZpVYkaiUoirWm4:-KYwpXZpVYkaiUoirWm5:b-p0muu8) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-001-inline.html#L22)):*
> ```HTML
> <div class="flex">
> P
>   <div class="contents">
> ```

I don't understand how this test can detect failures. If I change the div class="contents" elements to spans the (expected) output is still the same.

---

*[css-display-3/display-contents-dynamic-flex-001-none.html, line 26 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwqGTBhVaa6rUCttGn:-KYwqGTBhVaa6rUCttGo:b-m11zxh) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-001-none.html#L26)):*
> ```HTML
>   </div>
>   <div class="contents">
>     <div class="inline">S<div>0</div></div>
> ```

Why is the "0" div essential to this test?

---

*[css-display-3/display-contents-dynamic-flex-002-inline.html, line 14 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwxTKFD7uHqUriezTR:-KYwxTKFD7uHqUriezTS:b-guawpj) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-002-inline.html#L14)):*
> ```HTML
>   <div class="contents c1">x</div>
>   <div class="contents c1"><div class="contents c2">y</div></div>
>   <div class="contents c1"><div class="contents c2"><div class="">1<span class="b">1</span></div></div></div>
> ```

Why the empty class attributes?

---

*[css-display-3/display-contents-dynamic-flex-002-inline.html, line 24 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwxnJ8rCmX9lAzS4CS:-KYwxnJ8rCmX9lAzS4CT:bjf9j4a) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-002-inline.html#L24)):*
> ```HTML
>   <div class="contents c9"><div class="">8</div></div>
>   <div class="contents c9"><div class="contents">9<div class="contents c2">a<span class="b">b</span>c</div></div></div>
>   <div class="contents c10"><div class="">10</div></div>
> ```

In general this test looks overly complicated and really hard to understand what we're actually testing with all theses cases apart from the fact that we are testing the transition back and forth between display:inline and display:contents on flex items.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by emilio** [comment]
Review status: 4 of 39 files reviewed at latest revision, 6 unresolved discussions.

---

*[css-display-3/display-contents-computed-style.html, line 72 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwkfbGJvD0lzkhHBnq:-KYx-KMKL3XHuKtH97XD:bebrt1w) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-computed-style.html#L72)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

Add a spec link to https://drafts.csswg.org/css-display-3/#transformations and -> "display:contents is blockified for root elements"
</blockquote></details>

Will do

---

*[css-display-3/display-contents-dynamic-flex-001-inline.html, line 22 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwpXZpVYkaiUoirWm4:-KYx-7pIPO4sucAmICQz:b-ly5epd) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-001-inline.html#L22)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

I don't understand how this test can detect failures. If I change the div class="contents" elements to spans the (expected) output is still the same.
</blockquote></details>

The problem with this test was the transition from inline to contents again.

Due to bad style adjustment, we were not blockifying the inline element, which created more anonymous boxes, and bad layout on re-attachment as a result.

---

*[css-display-3/display-contents-dynamic-flex-001-none.html, line 26 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwqGTBhVaa6rUCttGn:-KYwzriJXf39VBjmmByz:b-gpu9v8) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-001-none.html#L26)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

Why is the "0" div essential to this test?
</blockquote></details>

Is a block in an inline, and that precise thing caught some re-attachment bugs in my implementation.

---

*[css-display-3/display-contents-dynamic-flex-002-inline.html, line 14 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwxTKFD7uHqUriezTR:-KYwzZ-vmHkqtW_w7Ycp:b1tqm1s) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-002-inline.html#L14)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

Why the empty class attributes?
</blockquote></details>

No reason, they where there in the original test, I'll strip them out.

---

*[css-display-3/display-contents-dynamic-flex-002-inline.html, line 24 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwxnJ8rCmX9lAzS4CS:-KYwzgD2GTVsy7MzANYJ:b-40l2e1) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-002-inline.html#L24)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

In general this test looks overly complicated and really hard to understand what we're actually testing with all theses cases apart from the fact that we are testing the transition back and forth between display:inline and display:contents on flex items.
</blockquote></details>

I agree, though this helped me to catch a few edge cases, so I'd rather leave it as-is.

---

*[css-display-3/display-contents-text-only-001.html, line 10 at r2](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KXqX0UyF_2vou0l3RcP-r2-10:-KYwzS_39sssnKRBYWwI:b-n8eezu) ([raw file](https://github.com/w3c/csswg-test/blob/3be5c4bd72336a0ab37547d0293d2463ddeb1060/css-display-3/display-contents-text-only-001.html#L10)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

I resolved it as blockification of contents on root element is specified in css-display-3.
</blockquote></details>

Thanks for doing that, missed that section. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by rune-opera** [comment]
*[css-display-3/display-contents-dynamic-flex-001-inline.html, line 22 at r5](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYwpXZpVYkaiUoirWm4:-KYx4SjiVNQSgJYp2KZs:bm1c00o) ([raw file](https://github.com/w3c/csswg-test/blob/7b2abaf831f07ffeef7f64695f1e1c65222c408e/css-display-3/display-contents-dynamic-flex-001-inline.html#L22)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary><blockquote>

The problem with this test was the transition from inline to contents again.

Due to bad style adjustment, we were not blockifying the inline element, which created more anonymous boxes, and bad layout on re-attachment as a result.
</blockquote></details>

Could you make the test so that you cover the case with anonymous box generation and still have a different rendering for inline and contents? Otherwise this test would not make sense for other vendors. Add a red border for the inline case for instance?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by emilio** [comment]
Added a red border to the display:contents element. Sorry for the delay, had to run yesterday for family reasons.

**Comment by rune-opera** [comment]
Reviewed 18 of 33 files at r1, 2 of 6 files at r3, 3 of 3 files at r5, 12 of 12 files at r6, 4 of 4 files at r7.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[css-display-3/display-contents-dynamic-multicol-001-inline.html, line 4 at r6](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYx88b1mwctlfCTigsC:-KYx88b2jg4bQ14Os98J:bnh4xp2) ([raw file](https://github.com/w3c/csswg-test/blob/8f5848639331141090621ebcc2eb95bcceebf61c/css-display-3/display-contents-dynamic-multicol-001-inline.html#L4)):*
> https://bugzilla.mozilla.org/show_bug.cgi?id=907396

This doesn't belong here. Could you contact Mats and hear in what form he wants his credits?

---

*[css-display-3/display-contents-dynamic-multicol-001-none.html, line 4 at r6](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYxA19RcJaVQhCmFPSl:-KYxA19SSHW3sqvCwtUU:b-1li3u5) ([raw file](https://github.com/w3c/csswg-test/blob/8f5848639331141090621ebcc2eb95bcceebf61c/css-display-3/display-contents-dynamic-multicol-001-none.html#L4)):*
> ```HTML
> <meta charset="utf-8">
> <title>CSS Test: CSS display:contents in multicolumn layout</title>
> <link rel="author" title="Mats Palmgren" href="https://bugzilla.mozilla.org/show_bug.cgi?id=907396">
> ```

Fix href.

---

*[css-display-3/display-contents-dynamic-table-001-inline.html, line 4 at r6](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYxA9FYZfn64mY6ojT5:-KYxA9FYZfn64mY6ojT6:b-1li3u5) ([raw file](https://github.com/w3c/csswg-test/blob/8f5848639331141090621ebcc2eb95bcceebf61c/css-display-3/display-contents-dynamic-table-001-inline.html#L4)):*
> ```HTML
> <meta charset="utf-8">
> <title>CSS Test: CSS display:contents</title>
> <link rel="author" title="Mats Palmgren" href="https://bugzilla.mozilla.org/show_bug.cgi?id=907396">
> ```

Fix href.

---

*[css-display-3/display-contents-dynamic-table-001-none.html, line 4 at r6](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYxAgkTb2HhPBltffOs:-KYxAgkTb2HhPBltffOt:b-1li3u5) ([raw file](https://github.com/w3c/csswg-test/blob/8f5848639331141090621ebcc2eb95bcceebf61c/css-display-3/display-contents-dynamic-table-001-none.html#L4)):*
> ```HTML
> <meta charset="utf-8">
> <title>CSS Test: CSS display:contents</title>
> <link rel="author" title="Mats Palmgren" href="https://bugzilla.mozilla.org/show_bug.cgi?id=907396">
> ```

Fix href.

---

*[css-display-3/display-contents-list-001-ref.html, line 17 at r7](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KZ6EA77Xu9JMSA1IHOx:-KZ6EA77Xu9JMSA1IHOy:b-as6tsz) ([raw file](https://github.com/w3c/csswg-test/blob/55cf4f9a9a8aba5ca27b97e9fdd75dd9b13a7f11/css-display-3/display-contents-list-001-ref.html#L17)):*
> ```HTML
>   <div class="inline"><div class="inline c4">4</div></div>
>   <div><div class="inline c5">5a</div></div>
>   <div class=" c5">5b</div>
> ```

unnecessary space in class attribute?

---

*[css-display-3/display-contents-text-only-001.html, line 8 at r7](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KZ6Fl_tUxSuBV5NzcM2:-KZ6Fl_tUxSuBV5NzcM3:b-pzz5qr) ([raw file](https://github.com/w3c/csswg-test/blob/55cf4f9a9a8aba5ca27b97e9fdd75dd9b13a7f11/css-display-3/display-contents-text-only-001.html#L8)):*
> ```HTML
> <link rel="match" href="display-contents-text-only-001-ref.html">
> <style>
>   html, body { display: contents; }
> ```

You can just drop "html" due to blockification?

---

*[css-display-3/support/acid.css, line 1 at r7](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KZ6GQdmY9iPkFi7uyrV:-KZ6GQdmY9iPkFi7uyrW:b210jzw) ([raw file](https://github.com/w3c/csswg-test/blob/55cf4f9a9a8aba5ca27b97e9fdd75dd9b13a7f11/css-display-3/support/acid.css#L1)):*
> ```CSS
> html, body {
> ```

Why is this file called acid? (apart from the fact that it was called that in the Gecko repo)

---

*[css-display-3/support/util.js, line 7 at r6](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYx75q6ZwtWtl5Na5J1:-KYx75q6ZwtWtl5Na5J2:b-sclpvv) ([raw file](https://github.com/w3c/csswg-test/blob/8f5848639331141090621ebcc2eb95bcceebf61c/css-display-3/support/util.js#L7)):*
> ```JavaScript
> 
>   // NOTE: Doing qsa('*') and getComputedStyle is just for the
>   // test's shake, since it's easier to mess it up when
> ```

The test's _shake_?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by emilio** [comment]
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[css-display-3/display-contents-dynamic-multicol-001-inline.html, line 4 at r6](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KYx88b1mwctlfCTigsC:-KZ6OT2do0b20FvZ-llb:bygbvad) ([raw file](https://github.com/w3c/csswg-test/blob/8f5848639331141090621ebcc2eb95bcceebf61c/css-display-3/display-contents-dynamic-multicol-001-inline.html#L4)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

> https://bugzilla.mozilla.org/show_bug.cgi?id=907396

This doesn't belong here. Could you contact Mats and hear in what form he wants his credits?
</blockquote></details>

FWIW this is how Mats landed this in mozilla-central: http://searchfox.org/mozilla-central/source/layout/reftests/css-display/display-contents-acid.html#10.

I'll contact him though.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by rune-opera** [comment]
Reviewed 13 of 13 files at r8.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by emilio** [comment]
So Mats replied, and he said that as long as the bug number was still referenced either from another link or from a comment he was fine with it.

I did the latter using:

```html
<!-- Imported from: https://bugzilla.mozilla.org/show_bug.cgi?id=907396 -->
```

**Comment by emilio** [comment]
I think the rest of the issues you raised are now fixed.

**Comment by rune-opera** [comment]
Reviewed 27 of 27 files at r9.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[css-display-3/support/acid.css, line 1 at r7](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-KZ6GQdmY9iPkFi7uyrV:-KZLGl96Pzrt2eo1Ht2X:b-bmuf2e) ([raw file](https://github.com/w3c/csswg-test/blob/55cf4f9a9a8aba5ca27b97e9fdd75dd9b13a7f11/css-display-3/support/acid.css#L1)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary><blockquote>

No other reason, happy to rename it to whatever you think is better.
</blockquote></details>

It's not that important

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155)*

**Comment by rune-opera** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1155#-:-KZLH4eBohkUpNpBL0Bd:bnfp4nl)*
